src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -162,41 +164,63 @@ IFACEMETHODIMP ZoneSet::AddZone(winrt::com_ptr<IZone> zone) noexcept
     return S_OK;
 }
 
-IFACEMETHODIMP_(winrt::com_ptr<IZone>)
-ZoneSet::ZoneFromPoint(POINT pt) noexcept
+IFACEMETHODIMP_(std::vector<int>)
+ZoneSet::ZonesFromPoint(POINT pt) noexcept
 {
-    winrt::com_ptr<IZone> smallestKnownZone = nullptr;
-    // To reduce redundant calculations, we will store the last known zones area.
-    int smallestKnownZoneArea = INT32_MAX;
-    for (auto iter = m_zones.rbegin(); iter != m_zones.rend(); iter++)
+    const int SENSITIVITY_RADIUS = 20;
+    std::vector<int> capturedZones;
+    for (size_t i = 0; i < m_zones.size(); i++)
+    {
+        auto zone = m_zones[i];
+        RECT newZoneRect = zone->GetZoneRect();
+        if (newZoneRect.left - SENSITIVITY_RADIUS <= pt.x && pt.x <= newZoneRect.right + SENSITIVITY_RADIUS &&
+            newZoneRect.top - SENSITIVITY_RADIUS <= pt.y && pt.y <= newZoneRect.bottom + SENSITIVITY_RADIUS &&
+            newZoneRect.left < newZoneRect.right && newZoneRect.top < newZoneRect.bottom) // proper zone
```

Comment:
We shouldn't, it's just a sanity check. Perhaps I should remove it?

Reply:
This unit test fails if this check is removed: `ZoneFromPointWithZeroRect`
What should I do?

Reply:
Sanity check are good, though they're usually checked as `assert`s, since if we actually have some invalid values, then we'd have problems all over the place! 

Regarding `ZoneFromPointWithNotNormalizedRect` and `ZoneFromPointWithZeroRect` tests, it looks like they're testing objects with invalid data. @SeraphimaZ, is it actually possible to have such zones? If those values invalid or a result of a bug, I think we should rather test that `MakeZone` throws/returns nullptr when provided with an invalid data.

Reply:
@yuyoyuppe regarding tests: since there is no validation in `MakeZone`, theoretically invalid zones are possible. I agree with you that validation is needed on zone creation

Reply:
Alright, but having such zones would be a bug, so let's add validation to `MakeZone`/`Zone` constructor and return `nullptr`/throw, and change the tests that they verify that we don't accept invalid values instead of trying to workaround them somehow. 

The sooner we know about bugs, the better, so I'd suggest do it as part of this PR.

Reply:
Sounds good but `MakeZone` is marked as `noexcept` and it's illegal to create a `com_ptr` from a null pointer

Reply:
It should be possible to create a `com_ptr` from nullptr, see https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/com-ptr#com_ptrcom_ptr-constructor

Reply:
If I do this, I get a crash in two unit tests which check behavior when zone spacing is large. This means that we sometimes do create invalid zones. I think the best thing to do is to just leave the code in this PR as-is, and take care of these zones later. Because I'm getting tired of chasing what happens when we disallow invalid zones while I'm working on something completely unrelated.

Reply:
alright, let's leave it as is. Perhaps we could create issues for
- invalid zones with large spacing
- adding validation to `MakeZone`
@enricogior

Reply:
@yuyoyuppe 
yes please open new issues.

Reply:
https://github.com/microsoft/PowerToys/issues/2003